### PR TITLE
EUI-4372 global search validations unhappy path

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-4372

### Change description ###
Enter values and click on Search - EUI data field format validations Unhappy path

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/search/containers/search-form/search-form.component.ts [hunk 1]
```diff
@@ -118,10 +129,111 @@ export class SearchFormComponent implements OnInit, OnDestroy {
...
+    this.formGroup.get(SearchFormControl.DATE_OF_DEATH_MONTH).setValidators(monthValidator);
+    yearValidator = SearchValidators.yearValidator();
+    this.formGroup.get(SearchFormControl.DATE_OF_DEATH_YEAR).setValidators(yearValidator);
+  }
+
+  /**
+   * Function to validate form controls
+   *
+   */
+  validateForm(): boolean {
```
Comment (Daniel-Lam) [inline src/search/containers/search-form/search-form.component.ts:170]: @johnbenjamin-hmcts - please mark function as either `public` or `private`.

File: src/search/containers/search-form/search-form.component.ts [hunk 2]
```diff
@@ -118,10 +129,111 @@ export class SearchFormComponent implements OnInit, OnDestroy {
...
+
+    // Validation succeeded, return true
+    return true;
+  }
+
+  /**
+   * Function to reset validation error messages
+   *
+   */
+  resetValidationErrorMessages(): void {
```
Comment (Daniel-Lam) [inline src/search/containers/search-form/search-form.component.ts:210]: @johnbenjamin-hmcts - please mark function as either `public` or `private`.

File: src/search/containers/search-form/search-form.component.ts [hunk 3]
```diff
@@ -118,10 +129,111 @@ export class SearchFormComponent implements OnInit, OnDestroy {
...
+    this.postcodeErrorMessage =
+    this.dateOfBirthErrorMessage =
+    this.dateOfDeathErrorMessage = null;
+  }
+
+  /**
+   * Function to check if any error exists
+   *
+   */
+  isAnyError(): boolean {
```
Comment (Daniel-Lam) [inline src/search/containers/search-form/search-form.component.ts:222]: @johnbenjamin-hmcts - please mark function as either `public` or `private`.

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - a few, very minor things to look at but otherwise all OK.

**Comment by johnbenjamin-hmcts** [comment]
> @johnbenjamin-hmcts - a few, very minor things to look at but otherwise all OK.

Thank you @Daniel-Lam, I have now applied the fixes.

**Review by Daniel-Lam — APPROVED** [review]

**Pull request merged**